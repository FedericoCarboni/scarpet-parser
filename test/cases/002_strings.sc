// String test
'plain string';
'escape sequences \n \t \' \\ plain text';
'string
spanning
multiple
lines'
