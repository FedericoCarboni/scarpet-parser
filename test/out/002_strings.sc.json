{
  "errors": [],
  "warnings": [],
  "root": {
    "kind": "BinaryExpression",
    "start": {
      "pos": 15,
      "row": 0,
      "col": 0
    },
    "end": {
      "pos": 106,
      "row": 5,
      "col": 0
    },
    "operator": ";",
    "lvalue": {
      "kind": "BinaryExpression",
      "start": {
        "pos": 15,
        "row": 0,
        "col": 0
      },
      "end": {
        "pos": 72,
        "row": 1,
        "col": 0
      },
      "operator": ";",
      "lvalue": {
        "kind": "StringLiteral",
        "start": {
          "pos": 15,
          "row": 0,
          "col": 0
        },
        "end": {
          "pos": 29,
          "row": 0,
          "col": 0
        },
        "value": "plain string"
      },
      "rvalue": {
        "kind": "StringLiteral",
        "start": {
          "pos": 31,
          "row": 1,
          "col": 0
        },
        "end": {
          "pos": 72,
          "row": 1,
          "col": 0
        },
        "value": "escape sequences \n \t ' \\ plain text"
      }
    },
    "rvalue": {
      "kind": "StringLiteral",
      "start": {
        "pos": 74,
        "row": 2,
        "col": 0
      },
      "end": {
        "pos": 106,
        "row": 5,
        "col": 0
      },
      "value": "string\nspanning\nmultiple\nlines"
    }
  }
}