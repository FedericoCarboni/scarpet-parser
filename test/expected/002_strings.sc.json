{
    "errors": [],
    "warnings": [],
    "root": {
        "kind": "BinaryExpression",
        "start": {
            "pos": 15,
            "row": 1,
            "col": 0
        },
        "end": {
            "pos": 106,
            "row": 6,
            "col": 6
        },
        "operator": ";",
        "lvalue": {
            "kind": "BinaryExpression",
            "start": {
                "pos": 15,
                "row": 1,
                "col": 0
            },
            "end": {
                "pos": 72,
                "row": 2,
                "col": 41
            },
            "operator": ";",
            "lvalue": {
                "kind": "StringLiteral",
                "start": {
                    "pos": 15,
                    "row": 1,
                    "col": 0
                },
                "end": {
                    "pos": 29,
                    "row": 1,
                    "col": 14
                },
                "value": "plain string"
            },
            "rvalue": {
                "kind": "StringLiteral",
                "start": {
                    "pos": 31,
                    "row": 2,
                    "col": 0
                },
                "end": {
                    "pos": 72,
                    "row": 2,
                    "col": 41
                },
                "value": "escape sequences \n \t ' \\ plain text"
            },
            "comment": "// String test\n"
        },
        "rvalue": {
            "kind": "StringLiteral",
            "start": {
                "pos": 74,
                "row": 3,
                "col": 0
            },
            "end": {
                "pos": 106,
                "row": 6,
                "col": 6
            },
            "value": "string\nspanning\nmultiple\nlines"
        },
        "comment": null
    }
}